### DISCUSSION TITLE
New FVM-focused developer testnets: Wallaby and Hyperspace

### DISCUSSION THREAD

**Discussion by raulk** [filecoin-project/FIPs#544] [General]
This is a small post to let the community know about two new developer-oriented Filecoin testnets.

## New testnets

* **[Wallaby](https://github.com/filecoin-project/testnet-wallaby) testnet** => the bleeding edge testnet, gets reset every 1-2 weeks with new FVM alpha releases from [incremental delivery](https://github.com/filecoin-project/ref-fvm/issues/692). Wallaby has been running for a few months now, already has 5 block explorers, and in the [HackFEVM](https://fevm.ethglobal.com/) and subsequent days it saw 2000+ contract deployments.
* **Hyperspace testnet** (formerly "Buildernet") => a new (yet unlaunched) developer testnet with a firmer **stability** guarantee. The aim is for this testnet to receive Beta-quality upgrades, and to minimise the amount of resets. This is important because devs don't appreciate having their progress wiped out from the network (which is what a reset does). We still expect some resets on irrecoverable events, but these should be minimised. Note that we are still figuring out the best way to conduct clean upgrades through Lotus, as the codebase doesn't support per-network diverging lineages very well at the moment.

## When to use each?

* Deploy on Wallaby when you want to test against the bleeding edge quasi-weekly releases that the FVM core team is putting out. Especially useful if you want to use a feature that has been recently introduced.
* Deploy on the upcoming Hyperspace testnet for longer-term testing efforts.

## Filecoin testnet positioning

|                                     	| Butterflynet                                                                                   	| Wallaby                                                                                                                                  	| Hyperspace                                                                                   	| Calibrationnet                                                                                                                                               	| Mainnet                                                        	|
|-------------------------------------	|------------------------------------------------------------------------------------------------	|------------------------------------------------------------------------------------------------------------------------------------------	|----------------------------------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------------------------------------------------------------	|----------------------------------------------------------------	|
| Description                         	| Post devnet, private Lotus development testnet                                                 	| Bleeding edge, resettable testnet for FVM development and future evolution                                                               	| Stable, reliable, perpetual developer testnet                                                	| Pre-production network                                                                                                                                       	| Production network                                             	|
| Mission and purpose                 	| First staging environment for Lotus releases with real mining (albeit with small sector sizes) 	| To serve as a playground to ship early FVM feature previews to the community                                                             	| To serve as the main pre-production developer testnet                                        	| Cross-implementation testnet; functionally equivalent to mainnet                                                                                             	| To be the Filecoin network                                     	|
| Scope                               	| Private testnet                                                                                	| Shared testnet                                                                                                                           	| Shared testnet                                                                               	| Shared testnet                                                                                                                                               	| Shared mainnet                                                 	|
| Longevity (*)                       	| On-demand                                                                                      	| Likely perpetual                                                                                                                         	| Perpetual                                                                                    	| Perpetual                                                                                                                                                    	| Perpetual                                                      	|
| Deployment technique                	| Resets                                                                                         	| Resets                                                                                                                                   	| Upgrades + reset as last resort on unrecoverable events, announced ahead of time if possible 	| Upgrades + reset as last resort on unrecoverable events, announced ahead of time if possible                                                                 	| Upgrades only                                                  	|
| Sector sizes                        	| 512 MiB, 32GiB, 64GiB                                                                          	| 512 MiB, 32 GiB, 64 GiB                                                                                                                  	| 512 MiB, 32 GiB, 64 GiB                                                                      	| 32 GiB, 64 GiB                                                                                                                                               	| 32 GiB, 64 GiB                                                 	|
| Minimum power                       	| 2GiB                                                                                           	| 16GiB                                                                                                                                    	| 16GiB                                                                                        	| 32GiB                                                                                                                                                        	| 1TiB                                                           	|
| Software quality / maturity level   	| Alpha                                                                                          	| Alpha                                                                                                                                    	| Beta                                                                                         	| RC                                                                                                                                                           	| Final                                                          	|
| Team ownership                      	| Lotus team                                                                                     	| FVM team                                                                                                                                 	| FVM team                                                                                     	| Core devs - implementers (heavily stewarded by Lotus)                                                                                                        	| N/A                                                            	|
| Operated by                         	| Lotus                                                                                          	| Factor8 Solutions (contract)                                                                                                             	| Factor8 Solutions (contract WIP)                                                             	| Fil-infra (via Lotus’ request)                                                                                                                               	| N/A                                                            	|
| Miner distribution                  	| Lotus TSEs + Protocol Labs + daring community members                                          	| Factor8 Solutions + Jim Pick                                                                                                             	| Factor8 Solutions + TBD                                                                      	| Global                                                                                                                                                       	| Global                                                         	|
| Known full nodes                    	| Protocol Labs + daring community members                                                       	| Factor8 Solutions + Jim Pick + Glif + Zondax (Beryl) + Filmine + Filfox                                                                  	| Factor8 Solutions + Jim Pick + Glif + Zondax (Beryl) + Filmine + Filfox                      	| Network participants                                                                                                                                         	| Everyone                                                       	|
| Stability expectation               	| None                                                                                           	| Medium                                                                                                                                   	| Very high                                                                                    	| Extreme                                                                                                                                                      	| Paranoid                                                       	|
| SLA                                 	| None                                                                                           	| Best-effort                                                                                                                              	| TBD                                                                                          	|                                                                                                                                                              	|                                                                	|
| Tolerance for catastrophic failures 	| High                                                                                           	| High, expectation of constant reset                                                                                                      	| Minimal, resets wipe out developer progress                                                  	| Minimal, resets wipe out valuable state and adds additional cost (engineering, devops, communication, testing, community operation) that will delay upgrades 	| Zero                                                           	|
| Deal making activity                	| Manual                                                                                         	| Requires manually finding an SP                                                                                                          	| Estuary, nft.storage, web3.storage, on manual request, potentially sample data (WIP)         	| User and application developer testing                                                                                                                       	| Organic                                                        	|
| Sector onboarding                   	| Manual                                                                                         	| Only initial sector onboarding on genesis, unless deals are requested                                                                    	| Continuous                                                                                   	| Continuous                                                                                                                                                   	| Continuous                                                     	|
| Proving pattern                     	| Manual                                                                                         	| Continuous success                                                                                                                       	| Miners with different faulting patterns (WIP)                                                	| Continuous                                                                                                                                                   	| Continuous                                                     	|
| FIL+ activity                       	| Manual                                                                                         	| None                                                                                                                                     	| Deals from aggregators are FIL+ endowed – need to ensure continuous datacap supply           	| Mainly within Lotus                                                                                                                                          	| Real                                                           	|
| Explorers                           	| None                                                                                           	| https://explorer.glif.io/ https://beryx.zondax.ch/ https://wallaby.filfox.info/ https://explorer.filmine.io/ https://wallaby.filscan.io/ 	| Expect same support as Wallaby                                                               	| https://calibration.filscan.io/ https://calibration.filscout.com/                                                                                            	| https://filfox.info/ https://filscout.com/ https://filscan.io/ 	|

(Thanks to the Lotus team for contributions to this table.)

**Comment by jennijuju** [comment]
Thanks for sharing @raulk!

The title suggests these two testnets are FVM-focused:
- Will these two network include all FIPs that are finalized/soon-to-be finalized in Mainnet, including the ones that do not require change to FVM or impact Filecoin smart contract/application developer?
- What level of testing do you think these two testnets can provide for built-in actors and client implementations?
- Can non-developers, i.e: storage providers, also join wallaby and hyperspace?

**Comment by trruckerfling** [comment]
@raulk another early builder Starboard has just launched their FEVM explorer for Wallaby, to add to the list! https://wallaby-fvm.starboard.ventures/contracts

**Comment by jimpick** [comment]
Small correction: Mainnet minimum power is 10TiB

**Comment by eshon** [comment]
@raulk - Wallaby and Hyperspace only support 512 MiB and 32 GiB miners

**Comment by jennijuju** [comment]
hyperspace is now live! for more details https://github.com/filecoin-project/testnet-hyperspace

**Comment by maciejwitowski** [comment]
## Moving to Calibrationnet as developer-focused testnet
FYI There has been a post published about sunsetting Hyperspace and moving to Calibrationnet at the end of May 2023 https://github.com/filecoin-project/community/discussions/631
